### PR TITLE
feat(projectEdit): Project creators and moderators can edit few fields in a closed project

### PR DESCRIPTION
**Pull request by smrutis1**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? 

Closes: #1044 

> * Did you add or update any new dependencies that are required for your change? - 

No

Issue:  #1044 

Add some user as moderator to a project. Make the project clearing state to closed. Login with user which is added as a moderator and edit the project. All fields should be disabled except editing external id, security vulnerability monitoring and security vulnerabilities display
> Have you implemented any additional tests? - No

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

Signed-off-by: Smruti Prakash Sahoo <smruti.sahoo@siemens.com>

### CODE REVIEW

**Review by akapti — CHANGES_REQUESTED** [review]
Tested the PR, and found following issues.

1. The `edit` button on project list page is still disabled for project creator and moderators.
2. `Security Vulnerability Monitoring` field is still disabled for non-admin users.
3. Modifying `External Id`, `Security Vulnerability Monitoring` and `Security Vulnerabilities Display` by project creator (or user with non-Admin role) is not being saved.
4. Project creator is getting added as `Moderator` whenever project is being saved (updated)

File: backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java [hunk 1]
```diff
@@ -616,12 +616,19 @@ private boolean duplicateAttachmentExist(Project project) {
...
     private boolean updateProjectAllowed(Project project, User user) {
         if (project.clearingState != null && project.clearingState.equals(ProjectClearingState.CLOSED)
-                && !PermissionUtils.isUserAtLeast(UserGroup.SW360_ADMIN, user)) {
+                && !PermissionUtils.isUserAtLeast(UserGroup.SW360_ADMIN, user) && !isModeratorOrCreator(project, user)) {
             return false;
         }
         return true;
     }
 
+    private boolean isModeratorOrCreator(Project project, User user) {
```
[the remark this answers is not in the document]
Comment (smrutis1) [reply backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java]: Moved it to `SW360Utils`

File: frontend/sw360-portlet/src/main/resources/META-INF/resources/html/projects/includes/projects/administrationEdit.jspf [hunk 2]
```diff
@@ -12,6 +12,7 @@
 <%@page import="org.eclipse.sw360.datahandler.thrift.projects.ProjectState"%>
 <%@page import="org.eclipse.sw360.datahandler.thrift.projects.Project" %>
 <core_rt:set var="isUnknownCTeamEnabled" value='<%=PortalConstants.IS_CLEARING_TEAM_UNKNOWN_ENABLED%>'/>
+<core_rt:set var="clearingStateClosedNUserNotAdmin" value="${(project.clearingState eq 'CLOSED') && (isUserAdmin != 'Yes')}"/>
```
[the remark this answers is not in the document]
Comment (smrutis1) [reply frontend/sw360-portlet/src/main/resources/META-INF/resources/html/projects/includes/projects/administrationEdit.jspf]: Done

**Comment by smrutis1** [comment]
Thanks for review and test. Have fixed the issues and regarding mentioned issue in 2nd point, the field will be enabled only when there is entries in the `Security Responsibles` filed.

**Comment by akapti** [comment]
Code looks good.
tested the PR, and is working as expected.

**Review by akapti — COMMENTED** [review]
Code looks good.
Tested the PR. Working as expected from UI. 
But Non-Admin user (creator & moderator) don't have `write` access to project via `REST` API.

**Comment by smrutis1** [comment]
@akapti , Please make sure `rest.write.access.usergroup=USER` is set in sw360.properties

**Review by akapti — APPROVED** [review]

**Pull request merged**